### PR TITLE
add calc function for waste energy use shares

### PR DESCRIPTION
**Pull request by fbenke-pik**

### CODE REVIEW

**Review by Renato-Rodrigues — CHANGES_REQUESTED** [review]
specific regional mapping aggregations should not be used inside of calc functions to keep support with multiple regional resolutions.

File: R/calcWasteEnergyUseShares.R [hunk 1]
```diff
@@ -1,64 +1,90 @@
...
   return(list(
-    x = out,
-    weight = weight,
+    x = numerator,
+    weight = NULL,
     unit = "%",
-    mixed_aggregation = TRUE,
+    aggregationFunction = .calcShares,
+    aggregationArguments = list(denominator = denominator),
+    min = 0, max = 1,
```
Comment (Renato-Rodrigues) [inline R/calcWasteEnergyUseShares.R:87]: Are the min and max parameters necessary?

**Review by Renato-Rodrigues — COMMENTED** [review]
Thank you for the changes added!
I don't have any additional suggestions, besides doing a last check for a given region to make sure the values makes sense on H12 and EU21 regional aggregations.

File: R/calcWasteEnergyUseShares.R (hunk 1 above)
Comment (fbenke-pik) [reply R/calcWasteEnergyUseShares.R:87]: No, it is just an optional check that makes sense to me, as these are shares and should always be between 0 and 1.

**Pull request merged**